skrub/_agg_joiner.py

```diff
@@ -145,44 +127,176 @@ class AggJoiner(BaseEstimator, TransformerMixin):
+        main_table_name="main_table",
+        aux_table_name="aux_table",
+    ):
+        """Check that there are no duplicate column names after aggregating.
+
+        Parameters
+        ----------
+        main_table : dataframe
+            The main table to join.
+        aux_table : dataframe
+            The aggregated auxiliary table to join.
+        main_table_name : str
+            How to refer to ``main_table`` in error messages.
+        aux_table_name : str
+            How to refer to ``aux_table`` in error messages.
+        Raises
+        ------
+        ValueError
+            If any of the table has duplicate column names, or if there are column
+            names that are used in both tables.
+        """
+        main_columns = list(set(main_table.columns) - set(main_key))
+        aux_columns = [f"{col}" for col in list(set(aux_table.columns) - set(aux_key))]
+        for columns, table_name in [
+            (main_columns, main_table_name),
+            (aux_columns, aux_table_name),
+        ]:
+            counts = Counter(columns)
+            duplicates = [k for k, v in counts.items() if v > 1]
+            if duplicates:
+                raise ValueError(
+                    f"Table '{table_name}' has duplicate column names: {duplicates}."
+                    " Please make sure column names are unique."
+                )
+        overlap = list(set(main_columns).intersection(aux_columns))
+        if overlap:
+            raise ValueError(
+                "After aggregating the auxiliary table, the following"
+                f" column names are found in both tables '{main_table_name}' and"
+                f" '{aux_table_name}': {overlap}. Please make sure column names do not"
+                " overlap by renaming some columns or choosing a different suffix."
+            )
+
+    def _check_inputs(self, X):
+        """Check inputs before fitting.
+
+        Parameters
+        ----------
+        X : DataframeLike
+            Input data, based table on which to left join the
+            auxiliary table.
+        """
+        X, self._aux_table = self._check_dataframes(X, self.aux_table)
+        self._main_key, self._aux_key = _join_utils.check_key(
+            self.main_key, self.aux_key, self.key
+        )
+        _join_utils.check_missing_columns(X, self._main_key, "'X' (the main table)")
+        _join_utils.check_missing_columns(self._aux_table, self._aux_key, "'aux_table'")
+
+        self._cols = atleast_1d_or_none(self.cols)
+        # If no cols provided, all columns but `aux_key` are used.
+        if len(self._cols) == 0:
+            self._cols = list(set(self._aux_table.columns) - set(self._aux_key))
+        _join_utils.check_missing_columns(self._aux_table, self._cols, "'aux_table'")
+
+        self._operation = self._check_operation()
+        self.num_operations, self.categ_operations = split_num_categ_operations(
+            self._operation
+        )
+        self._check_suffix()
```

Comment:
I have the impression some of those checks could benefit other estimators such as Joiner, InterpolationJoiner. in a follow-up PR we may want to move them to join_utils and use them in those other estimators